vmod/vmod_debug_obj.c

```diff
@@ -79,6 +80,29 @@ xyzzy_obj__fini(struct xyzzy_debug_obj **op)
 	FREE_OBJ(o);
 }
 
+VCL_INT v_matchproto_(td_xyzzy_objcli)
+xyzzy_objcli(VRT_CTX, struct xyzzy_debug_obj *o, VCL_STRANDS s)
+{
+	int i;
+
+	CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);
+	CHECK_OBJ_NOTNULL(o, VMOD_DEBUG_OBJ_MAGIC);
+
+	if (s->n > 0 && ! strcmp(s->p[0], "fail")) {
+		VSB_cat(ctx->msg, "You asked me to fail");
+		return (300);
+	}
+	VSB_cat(ctx->msg, o->vcl_name);
+	VSB_putc(ctx->msg, ':');
+
+	for (i = 0; i < s->n; i++) {
+		VSB_putc(ctx->msg, ' ');
+		VSB_cat(ctx->msg, s->p[i]);
+	}
+
+	return (200);
```

Comment:
Should we expose (or even restrict to) a subset of the VCLI status constants to VMODs?